config.layout

```diff
@@ -277,3 +277,23 @@
     logdir:        /var/log+
     cachedir:      /var/lib+
 </Layout>
+
+<Layout NixOS>
+    prefix:        /opt/trafficserver
+    exec_prefix:   ${prefix}
+    bindir:        ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/sbin
+    libdir:        ${exec_prefix}/lib
+    libexecdir:    ${exec_prefix}/libexec
+    infodir:       ${prefix}/share/info
+    mandir:        ${prefix}/share/man
+    sysconfdir:    /etc+
+    datadir:       ${prefix}/share+
+    docdir:        ${prefix}/share/doc+
+    installbuilddir: ${datadir}/build
+    includedir:    ${prefix}/include
+    localstatedir: /var/lib+
+    runtimedir:    /run+
+    logdir:        /var/log+
+    cachedir:      /var/cache+
+</Layout>
```

Comment:
About half of these are replaced through configure flags when building with Nix, but I still added them for the sake of consistency.